subsys/shell/backends/shell_uart.c

```diff
@@ -176,16 +177,19 @@ static void uart_tx_handle(const struct device *dev, struct shell_uart_int_drive
 		return;
 	}
 
-	len = ring_buf_get_claim(&sh_uart->tx_ringbuf, (uint8_t **)&data,
+	avail = ring_buf_get_claim(&sh_uart->tx_ringbuf, (uint8_t **)&data,
 				 sh_uart->tx_ringbuf.size);
-	if (len) {
+	if (avail) {
 		int err;
-
-		len = uart_fifo_fill(dev, data, len);
-		err = ring_buf_get_finish(&sh_uart->tx_ringbuf, len);
+		written = uart_fifo_fill(dev, data, avail);
+		err = ring_buf_get_finish(&sh_uart->tx_ringbuf, written);
 		__ASSERT_NO_MSG(err == 0);
 		ARG_UNUSED(err);
 	} else {
+		written = 0;
+	}
+
+	if (written == avail) {
```

Comment:
I am curious about the scenario where `written != avail`. In the case that there's no new data in the buffer, would this lead to another IRQ being called solely to disable the IRQ? I'm just trying to fully understand the flow here. Thanks for clarifying!

@nordic-krch: any thoughts?